core/internal/runhistoryreader/reader.go

```
@@ -226,19 +270,32 @@ func (h *HistoryReader) initParquetFiles(ctx context.Context) error {
 	for i, url := range signedUrls {
 		var parquetFile *pqarrow.FileReader
 
-		// When the user doesn't specify any keys,
-		// It is faster to download the entire parquet file
-		// and process it locally.
-		if len(h.keys) == 0 {
-			tmpDir := os.TempDir()
-			fileName := fmt.Sprintf("run_history_%d.parquet", i)
+		dir := os.Getenv("WANDB_CACHE_DIR")
+		if dir == "" {
+			dir, _ = os.UserCacheDir()
```

Should probably not ignore the error?